components/shortcut/actions/create-story/create-story.js

```diff
@@ -0,0 +1,357 @@
+        return await this.shortcut.listMembersAsOptions();
+      },
+      optional: true,
+    },
+    startedAtOverride: {
+      type: "string",
+      label: "Started at Override Date",
+      description: "A manual override for the time/date the Story was started.",
+      optional: true,
+    },
+    storyLink: {
+      type: "object",
+      label: "Story Link",
+      description:
+        "An story link object attached to the story. Each story link object must have the following structure: `object_id` which is an integer, unique ID of the story defined as object, `subject_id` which is an integer, unique ID of the story defined as subject, and `verb` which indicates how the subject story acts on the object story, valid values are `blocks`, `duplicates`, or `relates to`. See [CreateStoryLinkParams](https://shortcut.com/api/rest/v3#CreateStoryLinkParams) for more info.",
+      optional: true,
+    },
+    storyType: {
+      type: "string",
+      label: "Story Type",
+      description: "The type of story (feature, bug, chore).",
+      options: [
+        "bug",
+        "chore",
+        "feature",
+      ],
+      default: "feature",
+      optional: true,
+    },
+    task: {
+      type: "object",
+      label: "Task",
+      description:
+        "A task object attached to the story. Each task object must have the following structure: `complete` which is a boolean, indicating whether the task is completed (defaults to `false`), `created_at` which defaults to the time/date the task is created but can be set to reflect another creation time/date, `description` as a description for the task, `external_id` a field can be set to another unique ID. In the case that the task has been imported from another tool, the ID in the other tool can be indicated here, `owner_ids` as an array of UUIDs for any members you want to add as owners on this new task, `updated_at` which defaults to the time/date the task was last updated in Shortcut but can be set to reflect another time/date.  See [CreateTaskParams](https://shortcut.com/api/rest/v3#CreateTaskParams) for more info.",
+      optional: true,
+    },
+    updatedAt: {
+      type: "string",
+      label: "Updated at Date",
+      description: "The time/date the story was updated.",
+      optional: true,
+    },
+    workflowStateId: {
+      type: "integer",
+      label: "Workflow State Id",
+      description: "The ID of the workflow state the story will be in.",
+      async options() {
+        const workflows = await this.shortcut.callWithRetry("listWorkflows");
+        return workflows.reduce(function (options, workflow) {
+          const hasState = get(workflow, [
+            "states",
+            "length",
+          ]);
+          if (hasState) {
+            workflow.states.map((state) => (options.push({
+              label: `${state.name} (${workflow.name})`,
+              value: `${state.id}`,
+            })));
+          }
+          return options;
```

Comment:
You can also do it like this:
```suggestion
          if (!hasState) {
            return options;
          }
          const optionsToAdd = workflow.states.map((state) => ({
            label: `${state.name} (${workflow.name})`,
            value: `${state.id}`,
          }));
          return options.concat(optionsToAdd)
```

Reply:
OK.